### ISSUE TITLE
Pulumi cli crashes with error: confirmation cancelled, not proceeding with the update: Unexpected Escape Sequence: ['\x1b' 'O']

### ISSUE THREAD

**Issue by vladkosarev** [pulumi/pulumi#7071]
pulumi 3.2.1 cli crashes when trying to use arrow keys to make a selection
error: confirmation cancelled, not proceeding with the update: Unexpected Escape Sequence: ['\x1b' 'O']

## Expected behavior

pulumi cli should not crash

## Current behavior

```
 +   pulumi:pulumi:Stack      quickstart-dev        create     
 +   ├─ awsx:x:ec2:Vpc        default-vpc           create     
 +   │  ├─ awsx:x:ec2:Subnet  default-vpc-public-0  create     
 +   │  └─ awsx:x:ec2:Subnet  default-vpc-public-1  create     
 +   ├─ aws:rds:SubnetGroup   dbsubnets             create     
 +   └─ aws:rds:Instance      postgresdb            create     
 
Resources:
    + 6 to create

Do you want to perform this update?  [Use arrows to move, enter to select, type to filter]
  yes
> no
  details
error: confirmation cancelled, not proceeding with the update: Unexpected Escape Sequence: ['\x1b' 'O']
```
## Steps to reproduce

Just running 'pulumi up' breaks

## Context (Environment)

I am running under wsl2/bash/xterm-256color environment.
I tried a different terminal (alacrity) and that worked fine.
I have been using my environment for years and this is the first time a cli app broke like that.
I have read the other bug reports about this issue and they are all closed so the assumption seems to be that this issue is fixed.

**Comment by mark-bixler** [comment]
This issue just recently started for me. Running MacOS 12.6. 

```
╰─❯ pulumi about
CLI          
Version      3.46.1
Go Version   go1.19.3
Go Compiler  gc

Plugins
NAME    VERSION
nodejs  unknown

Host     
OS       darwin
Version  12.6
Arch     x86_64

This project is written in nodejs: executable='/usr/local/bin/node' version='v18.8.0'
```

**Comment by mark-bixler** [comment]
I can confirm this broke after installing `3.46.1` & continues to a problem through `3.47.2`. I have reverted back to `3.46.0` and do not have the problem. MacOS 12.6 / iTerm2 / zsh / xterm-256color.

**Comment by tmehlinger** [comment]
Similar issue for me on 3.51.1 and 3.55.0:
```
error: confirmation cancelled, not proceeding with the update: EOF
```

**Comment by justinvp** [comment]
This can happen when a `dotnet` command like `dotnet build` is run. See https://github.com/dotnet/sdk/issues/15243.

I believe this has now been fixed with https://github.com/pulumi/pulumi/pull/13979, which upgraded the library we use to do the prompt to a more recent version which includes [this fix](https://github.com/go-survey/survey/issues/341). This shipped in Pulumi [v3.85.0](https://github.com/pulumi/pulumi/releases/tag/v3.85.0). Therefore, I'm going to close this as fixed.

Please open a new issue if you're still seeing the problem with Pulumi v3.85.0 or higher.
